src/model/AirLoopHVAC.cpp

```diff
@@ -721,40 +721,101 @@ namespace model {
 
       airLoopClone.getImpl<detail::AirLoopHVAC_Impl>()->createTopology();
 
-      auto supplyComps = supplyComponents();
-      auto outletNodeClone = airLoopClone.supplyOutletNode();
-      std::vector<Node> supplyNodes;
+      struct HVACPath
+      {
+        HVACPath(const HVACComponent& t_originalStart, const HVACComponent& t_originalEnd, const HVACComponent& t_clonedStartOrEndOfPath,
+                 const Node& t_clonedAddNode, bool t_reverse)
+          : originalStart(t_originalStart),
+            originalEnd(t_originalEnd),
+            clonedStartOrEndOfPath(t_clonedStartOrEndOfPath),
+            clonedAddNode(t_clonedAddNode),
+            reverse(t_reverse){};
+
+        HVACComponent originalStart;
+        HVACComponent originalEnd;
+        HVACComponent clonedStartOrEndOfPath;  // If reverse is false, it's the start, if reverse is true, it's the end
+        Node clonedAddNode;
+        bool reverse;
+      };
 
-      for (const auto& comp : supplyComps) {
-        if (comp.iddObjectType() == Node::iddObjectType()) {
-          supplyNodes.push_back(comp.cast<Node>());
-        } else {
-          auto compClone = comp.clone(model).cast<HVACComponent>();
-          compClone.addToNode(outletNodeClone);
-          // If the original component was also on a PlantLoop
-          if (boost::optional<HVACComponent> hvacComp = comp.optionalCast<HVACComponent>()) {
-            if (boost::optional<PlantLoop> pl = hvacComp->plantLoop()) {
-              // Connect the clone to the plantLoop too
-              pl->addDemandBranchForComponent(compClone);
+      std::vector<HVACPath> pathsToHandle;
+
+      if (isDualDuct()) {
+        // Add the Splitter to make it a dual duct
+        auto supplyInletNodeClone = airLoopClone.supplyInletNode();
+        ConnectorSplitter supplySplitterClone(model);
+        supplySplitterClone.addToNode(supplyInletNodeClone);
+        airLoopClone.getImpl<detail::AirLoopHVAC_Impl>()->setSupplySplitter(supplySplitterClone);
+
+        auto supplyOutletNodeClones = airLoopClone.supplyOutletNodes();
+
+        auto supplySplitter = this->supplySplitter().get();
+        auto supplyInletNode = this->supplyInletNode();
+        auto supplyOutletNodes = this->supplyOutletNodes();
+
+        pathsToHandle.emplace_back(supplyInletNode, supplySplitter, supplySplitterClone, supplyInletNodeClone, true);
+        pathsToHandle.emplace_back(supplySplitter, supplyOutletNodes[0], supplySplitterClone, supplyOutletNodeClones[0], false);
+        pathsToHandle.emplace_back(supplySplitter, supplyOutletNodes[1], supplySplitterClone, supplyOutletNodeClones[1], false);
+
+      } else {
+
+        pathsToHandle.emplace_back(this->supplyInletNode(), this->supplyOutletNode(), airLoopClone.supplyInletNode(), airLoopClone.supplyOutletNode(),
+                                   false);
```

Comment:
Single Duct: everything from supply inlet to supply outlet in one go, adding always to supplyOutletNode so it's in the right order